### ISSUE TITLE
Broken Tweets

### ISSUE THREAD

**Issue by h4ng3r** [fennb/phirehose#71]
I just downloaded the last version today. Using the ghetto example the consumer doesn't get the full json tweet, it gets just a part. This is and example:

```
18026,"created_at":"Thu Mar 08 21:44:33 +0000 2012","utc_offset":7200,"time_zone":"Amsterdam","geo_enabled":false,"lang":"en","contributors_enabled":false,"is_translator":false,"profile_background_color":"C0DEED","profile_background_image_url":"http:\/\/abs.twimg.com\/images\/themes\/theme1\/bg.png","profile_background_image_url_https":"https:\/\/abs.twimg.com\/images\/themes\/theme1\/bg.png","profile_background_tile":false,"profile_link_color":"0084B4","profile_sidebar_border_color":"C0DEED","profile_sidebar_fill_color":"DDEEF6","profile_text_color":"333333","profile_use_background_image":true,"profile_image_url":"http:\/\/pbs.twimg.com\/profile_images\/509391853473775620\/dtSNKnQD_normal.jpeg","profile_image_url_https":"https:\/\/pbs.twimg.com\/profile_images\/509391853473775620\/dtSNKnQD_normal.jpeg","profile_banner_url":"https:\/\/pbs.twimg.com\/profile_banners\/518908009\/1410299354","default_profile":true,"default_profile_image":false,"following":null,"follow_request_sent":null,"notifications":null},"geo":null,"coordinates":null,"place":null,"contributors":null,"retweeted_status":{"created_at":"Wed Oct 01 07:58:12 +0000 2014","id":517221930462883840,"id_str":"517221930462883840","text":"Today\u2019s #competition Just favourite and retweet this tweet to be in with a chance of winning. Winner chosen at random tomorrow. GO!","source":"\u003ca href=\"https:\/\/about.twitter.com\/products\/tweetdeck\" rel=\"nofollow\"\u003eTweetDeck\u003c\/a\u003e","truncated":false,"in_reply_to_status_id":null,"in_reply_to_status_id_str":null,"in_reply_to_user_id":null,"in_reply_to_user_id_str":null,"in_reply_to_screen_name":null,"user":{"id":25534147,"id_str":"25534147","name":"Volvo Trucks UK ","screen_name":"VolvoTrucksUK","location":"Warwick, United Kingdom","url":"http:\/\/www.volvotrucks.co.uk","description":"Total Commercial Vehicle Solution Provider, providing quality Volvo Trucks within the UK market http:\/\/www.volvotrucks.co.uk","protected":false,"verified":false,"followers_count":12956,"friends_count":1114,"listed_count":102,"favourites_count":584,"statuses_count":12151,"created_at":"Fri Mar 20 16:55:44 +0000 2009","utc_offset":3600,"time_zone":"London","geo_enabled":false,"lang":"en","contributors_enabled":false,"is_translator":false,"profile_background_color":"FFFFFF","profile_background_image_url":"http:\/\/pbs.twimg.com\/profile_background_images\/676413883\/7b8aa0d3bee4d2f57b4da2cf10387585.jpeg","profile_background_image_url_https":"https:\/\/pbs.twimg.com\/profile_background_images\/676413883\/7b8aa0d3bee4d2f57b4da2cf10387585.jpeg","profile_background_tile":false,"profile_link_color":"2FC2EF","profile_sidebar_border_color":"FFFFFF","profile_sidebar_fill_color":"FFFFFF","profile_text_color":"0E083D","profile_use_background_image":true,"profile_image_url":"http:\/\/pbs.twimg.com\/profile_images\/441162209654108160\/rRINCdyl_normal.jpeg","profile_image_url_https":"https:\/\/pbs.twimg.com\/profile_images\/441162209654108160\/rRINCdyl_normal.jpeg","profile_banner_url":"https:\/\/pbs.twimg.com\/profile_banners\/25534147\/1405412142","default_profile":false,"default_profile_image":false,"following":null,"follow_request_sent":null,"notifications":null},"geo":null,"coordinates":null,"place":null,"contributors":null,"retweet_count":216,"favorite_count":107,"entities":{"hashtags":[{"text":"competition","indices":[8,20]}],"trends":[],"urls":[],"user_mentions":[],"symbols":[]},"favorited":false,"retweeted":false,"possibly_sensitive":false,"filter_level":"low","lang":"en"},"retweet_count":0,"favorite_count":0,"entities":{"hashtags":[{"text":"competition","indices":[27,39]}],"trends":[],"urls":[],"user_mentions":[{"screen_name":"VolvoTrucksUK","name":"Volvo Trucks UK ","id":25534147,"id_str":"25534147","indices":[3,17]}],"symbols":[]},"favorited":false,"retweeted":false,"possibly_sensitive":false,"filter_level":"medium","lang":"en","timestamp_ms":"1412234939794"}
```

For me, seems a problem with newlines because the files generated by the collector don't have any newline.

**Comment by rubytastic** [comment]
Same issue, the process.php does not seem to get the full json object.

It reads the content of the temp files with
		while ($rawStatus = fgets($fp, 8192)) {


This could never work since this just grabs the 8192 chars, A tweet can always be longer or shorter. At least it does not grabb the correct full tweet. I tried to change the lenght but this still fails. 

Perhaps saving 1 tweet per file could resolve issues with broken json

**Comment by fennb** [comment]
Fixed in 197b7da3d0b63f2de972fb2a14c838feb4f304dc
